tests/src/main/scala/scalaz/tests/MaybeTests.scala

```diff
@@ -0,0 +1,131 @@
+package scalaz
+package tests
+
+import Predef._
+import data._
+import laws._
+import Scalaz._
+
+import testz._
+import z._
+
+object MaybeTests {
+
+  def tests[T](harness: Harness[T]): T = {
+    import harness._
+
+    val maybes =
+      IList[Maybe[String]](
+        Maybe.empty,
+        Maybe.just("hello"),
+        Maybe.just("world"),
+        Maybe.just("foobar"),
+      )
+
+    section(
+      namedSection("concrete")(
+        namedSection("empty")(
+          test("is cached")(
+            () => assert(Maybe.empty.asInstanceOf[scala.AnyRef] eq Maybe.empty.asInstanceOf[scala.AnyRef])
+          ),
+          test("is `None`")(() => assert(Maybe.empty.asInstanceOf[scala.AnyRef] eq scala.None)),
+        ),
+        test("just is Some")(() => assert(Maybe.just(1) == scala.Some(1))),
+        test("toOption is identity")(
+          () =>
+            maybes.foldMap { m =>
+              assert(m.asInstanceOf[scala.AnyRef] eq Maybe.toOption(m))
+          }
+        ),
+      ),
+      namedSection("laws")(
+        namedSection("eq laws")(
+          test("reflexivity") { () =>
+            maybes.foldMap(
+              EqLaws.reflexivity(_)(assert)
+            )
+          },
+          test("identity") { () =>
+            maybes.cross(maybes).foldMap {
+              case (m1, m2) => assert((m1 === m2) == (m1 == m2))
```

Comment:
This is not a very good test for `Eq`, but I'll fix all `Eq` laws later.

Reply:
It is insufficient because it does not test that `m1 === m2` does not compare the `a: A` inside of maybes using `==`.